nqeit-iceberg-dremio-extension/src/main/java/org/projectnessie/integtests/dremio/DremioHelper.java

```diff
@@ -0,0 +1,133 @@
+import org.apache.hc.client5.http.classic.methods.HttpUriRequestBase;
+import org.apache.hc.client5.http.impl.classic.CloseableHttpClient;
+import org.apache.hc.client5.http.impl.classic.HttpClients;
+import org.apache.hc.core5.http.io.entity.EntityUtils;
+import org.apache.hc.core5.http.io.entity.StringEntity;
+import org.json.JSONArray;
+import org.json.JSONObject;
+
+public class DremioHelper {
+  private String token;
+  private String projectId;
+  private String baseUrl;
+  private String catalogName;
+
+  DremioHelper(String projectId, String token, String baseUrl, String catalogName) {
+    this.projectId = projectId;
+    this.token = token;
+    this.baseUrl = baseUrl;
+    this.catalogName = catalogName;
+  }
+
+  private String createPayload(String query) {
+    String payload =
+        format("{ \"sql\": \"%s\", \"context\": [\"%s\", \"db\"] }", query, catalogName);
+    return payload;
+  }
+
+  private String executeHttpRequest(HttpUriRequestBase req) {
+    req.addHeader("Authorization", "Bearer " + token);
+    req.addHeader("Content-Type", "application/json");
+    try (CloseableHttpClient httpClient = HttpClients.createDefault()) {
+      return httpClient.execute(req, response -> EntityUtils.toString(response.getEntity()));
+    } catch (IOException e) {
+      throw new UncheckedIOException(e);
+    }
+  }
+
+  private String executeHttpPost(String url, String payload) {
+    HttpPost httpPost = new HttpPost(url);
+    httpPost.setEntity(new StringEntity(payload));
+    return executeHttpRequest(httpPost);
+  }
+
+  private String executeHttpGet(String url) {
+    HttpGet httpGet = new HttpGet(url);
+    return executeHttpRequest(httpGet);
+  }
+
+  private String waitForJobStatus(String jobId) {
+    String jobState = "RUNNING";
+    String url = baseUrl + "/v0/projects/" + projectId + "/job/" + jobId;
+    Set<String> finalJobStatesList = new HashSet<>(asList("COMPLETED", "FAILED", "CANCELED"));
+    while (!finalJobStatesList.contains(jobState)) {
+      String result = executeHttpGet(url);
+      JSONObject jObject = new JSONObject(result);
+      jobState = jObject.getString("jobState");
+    }
+    return jobState;
+  }
+
+  private List<List<Object>> parseQueryResult(String jobId) {
+
+    String url = baseUrl + "/v0/projects/" + projectId + "/job/" + jobId + "/results";
+    String result = executeHttpGet(url);
+
+    JSONObject jObject = new JSONObject(result);
+    int noOfRows = jObject.getInt("rowCount");
+    final JSONArray table_data = jObject.getJSONArray("rows");
+    List<List<Object>> list = new ArrayList<>();
+    for (int i = 0; i < noOfRows; i++) {
+      final JSONObject row = table_data.getJSONObject(i);
+      list.add(asList(row.getInt("id"), row.getString("val")));
+    }
+    return list;
+  }
+
+  private String submitQueryAndGetJobId(String query) {
+    String payload = createPayload(query);
+    String url = baseUrl + "/v0/projects/" + projectId + "/sql";
```

Comment:
Please add a link to the web page describing the API

Reply:
done.